src/main/scala/com/lightbend/conductr/sbt/BundlePlugin.scala

```diff
@@ -395,93 +424,81 @@ object BundlePlugin extends AutoPlugin {
   }
 
   private def getConfig(config: Configuration, forAllSettings: Boolean): Def.Initialize[Task[String]] = Def.task {
-    val checkComponents = (checksConfigName in config).value match {
-      case (value, configName) if (forAllSettings || configName.isDefined) && value.nonEmpty =>
-        val checkInitialDelayValue = (checkInitialDelay in config).value
-        val checkInitialDelayInSeconds =
-          if (checkInitialDelayValue.toMillis % 1000 > 0)
-            checkInitialDelayValue.toSeconds + 1 // always round up
-          else
-            checkInitialDelayValue.toSeconds
-        Seq(
-          value.map(uri => s""""$uri"""").mkString(
-            s"""components = {
-                |  ${(normalizedName in config).value}-status = {
-                |    description      = "Status check for the bundle component"
-                |    file-system-type = "universal"
-                |    start-command    = ["check", "--initial-delay", "$checkInitialDelayInSeconds", """.stripMargin,
-            ", ",
-            s"""]
-                |    endpoints        = {}
-                |  }
-                |}""".stripMargin
-          )
-        )
-      case _ =>
-        Seq.empty[String]
-    }
-
-    def formatValue[T](format: String, valueAndConfigName: (T, Option[String])): Seq[String] =
+    def hasValue[T](valueAndConfigName: (T, Option[String])): Boolean =
       valueAndConfigName match {
-        case (value, configName) if forAllSettings || configName.isDefined => Seq(format.format(value))
-        case _                                                             => Seq.empty[String]
+        case (value, configName) if forAllSettings || configName.isDefined => true
+        case _                                                             => false
```

Comment:
This test is about determining if a value has been provided at a specific level of configuration, as distinct from delegation. We don’t want to determine the value of a given setting, but whether it has been defined at an exact point on the config axis…

For example:

lazy val BackendRegion = config("backend-region").extend(Bundle)
BundlePlugin.bundleSettings(BackendRegion)
inConfig(BackendRegion)(Seq(
  normalizedName := "reactive-maps-backend-region",
  BundleKeys.configurationName := (normalizedName in BackendRegion).value,
  ...
))

That test will return true for normalizedName only if it has been defined at the BackendRegion config level. Thus we know that we need to write the value out in our bundle.

In the case of forAllSettings then we don’t care - we’re going to write the value out.

Does this make sense?

> On 25 May 2016, at 3:48 PM, Mirco Dotta <notifications@github.com> wrote:
> 
> In src/main/scala/com/lightbend/conductr/sbt/BundlePlugin.scala <https://github.com/typesafehub/sbt-conductr/pull/156#discussion_r64518241>:
> 
> >        valueAndConfigName match {
> > -        case (value, configName) if forAllSettings || configName.isDefined => Seq(format.format(value))
> > -        case _                                                             => Seq.empty[String]
> > +        case (value, configName) if forAllSettings || configName.isDefined => true
> > +        case _                                                             => false
> Much better :) Though, if forAllSettings is true, the config name can be missing. Out of curiosity, why is that? :)
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/typesafehub/sbt-conductr/pull/156/files/f17e4d13bd5905f96593d5698f35018270b7a388#r64518241>